app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,52 @@
+module SessionsHelper
+  def login(user)
+    session[:user_id] = user.id
+  end
+
+  def current_user?(user)
+    current_user == user
+  end
+
+  def current_user
+    if (user_id = User.find_by(id: session[:user_id]))
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(cookies[:remember_token])
+        login user
+        @current_user = user
+      end
+    end
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def logout
+    forget(current_user)
+    @current_user = nil
+    session.delete(:user_id)
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
+  def redirect_back_or(default)
+    redirect_to(session[:forwarding_url] || default)
```

Comment:
There seems to be a problem with this line. When I try to login this is what happens:

![image](https://user-images.githubusercontent.com/6015315/57563942-e98c6a80-73d6-11e9-8a1b-4dceeb69a0b0.png)